src/Compiler/Driver/GraphChecking/Graph.fs

```diff
@@ -0,0 +1,107 @@
+﻿namespace FSharp.Compiler.GraphChecking
+
+#nowarn "1182"
+#nowarn "40"
+
+open System.Collections.Concurrent
+open System.Collections.Generic
+open System.Text
+open FSharp.Compiler.IO
+
+/// <summary> Directed Acyclic Graph (DAG) of arbitrary nodes </summary>
+type internal Graph<'Node> = IReadOnlyDictionary<'Node, 'Node[]>
+
+module internal Graph =
```

Comment:
Please more unit tests for this module

Reply:
Some tests are now in https://github.com/dotnet/fsharp/blob/3c9fa6b88299d8e3ba343e3928ea82b24fac090e/tests/FSharp.Compiler.ComponentTests/Miscellaneous/GraphTests.fs#L1